src/aws/region.rs

```diff
@@ -37,9 +33,8 @@ impl RegionOrEndpoint {
         }
     }
 
-    pub fn endpoint(&self) -> crate::Result<Option<Endpoint>> {
-        let uri = self.endpoint.as_deref().map(Uri::from_str).transpose()?;
-        Ok(uri.map(Endpoint::immutable))
+    pub fn endpoint(&self) -> Option<String> {
+        self.endpoint.clone()
```

Comment:
Curious, was this just something that was noticed that could be simplified or was there other motivations for changing that?

Reply:
`Endpoint` was deprecated and the related functions were changed to take just Strings.